mypy/checkexpr.py

```diff
@@ -185,7 +185,7 @@ def visit_call_expr(self, e: CallExpr, allow_none_return: bool = False) -> Type:
         """Type check a call expression."""
         if e.analyzed:
             # It's really a special form that only looks like a call.
-            return self.accept(e.analyzed, self.type_context[-1])
+            return self.accept(e.analyzed, self.type_context[-1], always_allow_any=True)
```

Comment:
I'm not totally sure. I removed it and none of the tests failed, so I am going to remove it. I also played around with it a little bit and didn't find any difference.

I did some investigation what it could be and I think it was me trying to allow `cast(Any, foo)`. 
But since then we decided to not allow that.